packages/jaeger-ui/src/components/TracePage/TraceStatistics/tableValues.tsx

```diff
@@ -45,21 +44,47 @@ function getChildOfSpans(parentID: string, allSpans: Span[]): Span[] {
 
 function computeSelfTime(span: Span, allSpans: Span[]): number {
   if (!span.hasChildren) return span.duration;
-  // We want to represent spans as half-open intervals like [startTime, startTime + duration).
-  // This way the subtraction preserves the right boundaries. However, DRange treats all
-  // intervals as inclusive. For example,
-  //       range(1, 10).subtract(4, 8) => range([1, 3], [9-10])
-  //       length=(3-1)+(10-9)=2+1=3
-  // In other words, we took an interval of length=10-1=9 and subtracted length=8-4=4.
-  // We should've ended up with length 9-4=5, but we got 3.
-  // To work around that, we multiply start/end times by 10 and subtract one from the end.
-  // So instead of [1-10] we get [10-99]. This makes the intervals work like half-open.
-  const spanRange = new DRange(10 * span.startTime, 10 * (span.startTime + span.duration) - 1);
-  const children = getChildOfSpans(span.spanID, allSpans);
-  children.forEach(child => {
-    spanRange.subtract(10 * child.startTime, 10 * (child.startTime + child.duration) - 1);
-  });
-  return Math.round(spanRange.length / 10);
+
+  let spanSelfTime = span.duration;
+  let previousSpanEndTime = span.startTime;
+
+  const children = getChildOfSpans(span.spanID, allSpans).sort((a, b) => a.startTime - b.startTime);
+
+  const parentSpanEndTime = span.startTime + span.duration;
+
+  for (let index = 0; index < children.length; index++) {
+    const child = children[index];
+
+    const spanEndTime = child.startTime + child.duration;
+    const spanStartsAfterParentEnded = child.startTime > parentSpanEndTime;
+    const spanEndsBeforePreviousSpan = spanEndTime < previousSpanEndTime;
+    if (spanStartsAfterParentEnded || spanEndsBeforePreviousSpan) {
+      continue;
+    }
+
+    let nonOverlappingStartTime = child.startTime;
+    let nonOverlappingDuration = child.duration;
+
+    const spanStartsBeforePreviousSpanEnds = child.startTime < previousSpanEndTime;
+    if (spanStartsBeforePreviousSpanEnds) {
```

Comment:
can we add add diagrams to illustrate the scenarios? something like
```
// child(n-1):  |.......|
// child(n):         |......|
```

Reply:
WDYT about using format like this? Also see my previous comments, looks like example for L59,L60 will be also useful.

```
parent |.....................|
child    |.......|
child        |.....|                  - spanStartsBeforePreviousSpanEnds is true
child                |.....|          - spanStartsBeforePreviousSpanEnds is false
```

And example for L75

```
parent |......................|
child                      |.....|        - last span included in self time calculation
child                       |.........|   - skipped
```

Reply:
yes, in every if() there is an implied configuration which could be represented by a diagram like that to make the logic easier to follow.

Reply:
I added the comments in 7b7286d2eaf8bdda84b97170cb575010169efd88